docs/dev/devel-env.rst

```diff
@@ -286,12 +294,152 @@ Also, run the **integration tests** for ListenBrainz.
 
-FAQ's
+Set up Listenbrainz Spark development environment
+=================================================
+
+The Listenbrainz Spark module is used to generate recommendations and stats using Apache Spark. The recommendations are generated based on collaborative filtering technique.
+
+To contribute to the ListenBrainz Spark project, you need a development environment.
+With your development environment, you can test your changes before submitting a
+patch back to the project. This guide helps you set up a development environment
+and run ListenBrainz Spark locally on your workstation. By the end of this guide, you
+will have…
+
+* Install system dependencies
+* Run ListenBrainz Spark locally on your workstation
+
+
+Install dependencies
+--------------------
+
+The ``listenbrainz_spark-server`` communicates with the ``listenbrainz-server``. 
+So you will need to set up the ``listenbrainz-server`` using the above-given steps.
+
+The ``listenbrainz_spark-server`` is shipped in Docker containers just like the 
+``listenbrainz-server``. This helps create your development environment and later deploy the application. Therefore, to work on the project, you need to install Docker
+and use containers for building the project. Containers save you from installing all of this on your own workstation.
+
+For different Docker installation instructions for your distribution refer the steps given above.
+
+Update config.py
+^^^^^^^^^^^^^^^^
+ 
+For accessing various features available you will need to update the ListenBrainz Spark configuration file. If this is your first time configuring ListenBrainz Spark, copy the sample to a live
+configuration.
+
+.. code-block:: bash
+
+    cp listenbrainz_spark/config.py.sample listenbrainz_spark/config.py
+
+
+Initialize ListenBrainz Spark containers
+----------------------------------------
+
+Next, run ``develop.sh spark build`` in the root of the repository. Using
+``docker-compose``, it creates multiple Docker containers for the different
+services and parts of the ListenBrainz Spark server. This script starts Hadoop,
+Spark, and the Listenbrainz Playground containers. This also makes it easy to stop
+them all later.
+
+The first time you run it, it downloads and creates the containers. But it's not
+finished yet.
+
+.. code-block:: bash
+
+    ./develop.sh spark build
+    
+
+Format the NameNode container
+-----------------------------
+
+If it's your first time initializing the containers, format the ``namenode`` container
+using the given command.
+
+.. code-block:: bash
+
+    ./develop.sh spark run --rm hadoop-master hdfs namenode -format -nonInteractive -force
+
+This will format the ``namenode`` and link the ``namenode`` and ``datanode`` to a single ClusterID. 
+
+Remember that you shouldn't execute this command after running it for the first time. If run again it will
+format the ``namenode`` again and its link with the ``datanode`` will be lost, as both of them will have different
```

Comment:
But the user may want to format namenode. Instead of *should'nt execute* a warning stating that it will erase all the namenode data and its link with datanode will be lost.... will look good.

Reply:
Yeah. Makes more sense